aspnetcore/data/ef-rp/intro/samples/cu50/Pages/Students/Index.cshtml.cs

```diff
@@ -0,0 +1,29 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.Mvc;
+using Microsoft.AspNetCore.Mvc.RazorPages;
+using Microsoft.EntityFrameworkCore;
+using ContosoUniversity.Data;
+using ContosoUniversity.Models;
+
+namespace ContosoUniversity.Pages.Students
+{
+    public class IndexModel : PageModel
+    {
+        private readonly ContosoUniversity.Data.SchoolContext _context;
+
+        public IndexModel(ContosoUniversity.Data.SchoolContext context)
+        {
+            _context = context;
+        }
+
+        public IList<Student> Student { get;set; }
+
+        public async Task OnGetAsync()
+        {
+            Student = await _context.Students.Take(100).ToListAsync();
```

Comment:
@serpent5  how do I replace `.Take(100)` with `.Take(MvcOptions.MaxModelBindingCollectionSize ` ?

Reply:
You can inject `IOptions<MvcOptions>` into the constructor and access its `Value` property:

```csharp
private readonly MvcOptions _mvcOptions;

public IndexModel(IOptions<MvcOptions> mvcOptions)
{
    _mvcOptions = mvcOptions.Value;
}

public void OnGet()
{
    _mvcOptions.MaxModelBindingCollectionSize
}
```

The default for that seems to be `1024`, so I dunno if that's gonna be what you really want here, though...